src/utils/validate-schema.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Checks if Valid default coordinates
+ * @param {unknown} possibleCoordinates
+ * @param {number} dimensions - 2 for 2D Vector or 3 for 3D vector
+ * @returns {boolean} A boolean determining if coordinates are parsed correctly.
+ */
+function isValidDefaultCoordinate (possibleCoordinates, dimensions) {
+  if (typeof possibleCoordinates !== 'object' || possibleCoordinates === null) {
+    return false;
+  } else if (Object.keys(possibleCoordinates).length !== dimensions) {
+    return false;
+  } else {
+    if (dimensions === 2 && (possibleCoordinates.x === 0 || possibleCoordinates.x) && (possibleCoordinates.y === 0 || possibleCoordinates.y)) {
+      if (typeof possibleCoordinates.x === 'number' && typeof possibleCoordinates.y === 'number') {
+        return true;
+      }
+    }
+    if (dimensions === 3 && (possibleCoordinates.x === 0 || possibleCoordinates.x) && (possibleCoordinates.y === 0 || possibleCoordinates.y) && (possibleCoordinates.z === 0 || possibleCoordinates.z)) {
+      if (typeof possibleCoordinates.x === 'number' && typeof possibleCoordinates.y === 'number' && typeof possibleCoordinates.z === 'number') {
+        return true;
+      }
+    }
+    if (dimensions === 4 && (possibleCoordinates.x === 0 || possibleCoordinates.x) && (possibleCoordinates.y === 0 || possibleCoordinates.y) && (possibleCoordinates.z === 0 || possibleCoordinates.z) && (possibleCoordinates.w === 0 || possibleCoordinates.w)) {
+      if (typeof possibleCoordinates.x === 'number' && typeof possibleCoordinates.y === 'number' && typeof possibleCoordinates.z === 'number' && typeof possibleCoordinates.w === 'number') {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+module.exports.isValidDefaultCoordinate = isValidDefaultCoordinate;
+
+/**
+ * Validates the default values in a schema
+ * @param {string} type - type in a prop
+ * @param {unknown} defaultVal - default in a prop
+ * @returns {boolean} A boolean determining if defaults are valid.
+ */
+function isValidDefaultValue (type, defaultVal) {
```

Comment:
Yeah. Was going to unit test it afterwards since it's quite a bit of tests to write. But I'll get on it.